### ISSUE TITLE
Request path contains unescaped characters and GET 2.0/clusters/list loops

### ISSUE THREAD

**Issue by fpousserot** [paiqo/Databricks-VSCode#99]
Bonjour,

 I'm using the extension in databricks.connectionManager mode, with a working databricks cli, but the extension loops on the call to get clusters list and the extension returns Request path contains unescaped characters

do you know how find a solution a this context ? 

Merci

**Context :** 
Version: 1.71.2 (user setup)
Commit: 74b1f979648cc44d385a2286793c226e611f59e7
Date: 2022-09-14T21:03:37.738Z
Electron: 19.0.12
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
OS: Windows_NT x64 10.0.19044
Sandboxed: No
Databricks-VSCode : v1.1.4
Os : Windows 10
Network : behind entreprise proxy and intercept ssl certificat.
Python : 3.8
Environnement variable (user) : 
```
HTTPS_PROXY = http://proxy:80
NODE_EXTRA_CA_CERTS = D:\cacert.pem
REQUESTS_CA_BUNDLE = D:\cacert.pem
NODE_TLS_REJECT_UNAUTHORIZED = 0
```

databricks cli : ok (tested with python 3.8 and databricks clusters list)

vscode settings.json
```
"databricks.connectionManager": "Databricks CLI Profiles",
"databricks.lastActiveConnection": "DEFAULT",
"http.proxySupport": "off",
"http.proxyStrictSSL": false,
"http.proxy": "http://proxy:80",
```

.databrickscfg
```
[DEFAULT]
host=https://adb-xxxxxxx.x.azuredatabricks.net
token=dapixxxxxxxxxxxxxx
jobs-api-version=2.0

[xxxx-dev]
host=https://adb-xxxxxxx.x.azuredatabricks.net
token=dapixxxxxxxxxxxxxxxx
jobs-api-version=2.0
usecodecells=true
insecure=True
```

output vscode
```
Initializing ConnectionManager CLI ...
Configuration DEFAULT: Property "localSyncSubfolders" was not provided - using the default value!
Configuration DEFAULT: Property "exportFormats" was not provided - using the default value!
Configuration DEFAULT: Property "useCodeCells" was not provided - using the default value "true"!
Configuration claudia-dev: Property "localSyncSubfolders" was not provided - using the default value!
Configuration claudia-dev: Property "exportFormats" was not provided - using the default value!
Configuration claudia-dev: Property "useCodeCells" was not provided - using the default value "true"!
Setting 'databricks.lastActiveConnection' to 'DEFAULT' ...
Activating Databricks Connection 'DEFAULT' ...
Initializing Databricks API Service ...
Strict Proxy SSL verification disabled due to setting "http.proxyStrictSSL": false !
Proxy support is disabled due to setting "http.proxySupport": "off"!
Testing new Databricks API (https://adb-xxxxxxx.x.azuredatabricks.net) settings ...
GET 2.0/workspace/list
Params:{"params":{"path":"/"}}
Storing Personal Access Token in System KeyChain 'DEFAULT-API-Token' ...
Storing Personal Access Token in System KeyChain 'xxxx-dev-API-Token' ...
GET 2.0/clusters/list
Params:null
GET 2.0/clusters/list
Params:null
GET 2.0/clusters/list
Params:null
GET 2.0/clusters/list
Params:null
GET 2.0/clusters/list
Params:null
GET 2.0/clusters/list
Params:null
GET 2.0/clusters/list
Params:null
GET 2.0/clusters/list
Params:null
GET 2.0/clusters/list
Params:null
GET 2.0/clusters/list
Params:null
GET 2.0/clusters/list
Params:null
GET 2.0/clusters/list
Params:null
GET 2.0/clusters/list
Params:null
GET 2.0/clusters/list
Params:null
GET 2.0/clusters/list
Params:null
GET 2.0/clusters/list
Params:null
GET 2.0/clusters/list
Params:null
GET 2.0/clusters/list
Params:null
GET 2.0/clusters/list
Params:null
GET 2.0/clusters/list
Params:null```

**Comment by gbrueckl** [comment]
Thanks for reporting - there is definitely something wrong!
can you try to add `localsyncfolder = C:\` to the CLI config file?
it is actually a mandatory parameter (as of now) and you should get another error message
There have been some recent changes though and the CLI Connection Manager has not been thoroughly tested yet 

Will try to reproduce the issue on my side and then come back to you!

PS: you see these `GET 2.0/clusters/list` messages as the cluster browser updates every 30 seconds (used to be 10 in older versions)

**Comment by gbrueckl** [comment]
ok, I think I found the issue and I could also fix it
will be part of next release!

**Comment by fpousserot** [comment]
note : I try to add localsyncfolder = C:\ to the CLI config file , with the same result..

**Comment by yuanhao-li** [comment]
when the next release? many thanks.  🙏

**Comment by gbrueckl** [comment]
the issue should be fixed in this release, can you please try it and report back - thanks!
https://github.com/paiqo/Databricks-VSCode/releases/tag/v1.2.0

Note: it is not yet published to the VSCode gallery

**Comment by gbrueckl** [comment]
just published v1.2.1 to the gallery also 
closing this as fixed
